File: Cpp/2525_categorize_box_according_to_criteria.cpp
// https://leetcode.cn/problems/categorize-box-according-to-criteria/

class Solution {
public:
    string categorizeBox(int length, int width, int height, int mass) {
        bool bulky = false, heavy = false;
        int L = 1e4, V = 1e9, M = 100;
        if (length >= L || width >= L || height >= L || length * width >= V / height) {
            bulky = true;
        }
        if (mass >= M) {
            heavy = true;
        }
        if (bulky && heavy) return (string)"Both";
        else if (bulky && (!heavy)) return (string)"Bulky";
        else if ((!bulky) && heavy) return (string)"Heavy";
        else return (string)"Neither";
    }
};

File: Python3/2525_categorize_box_according_to_criteria.py
# https://leetcode.cn/problems/categorize-box-according-to-criteria/

class Solution:
    def categorizeBox(self, length, width, height, mass):
        """
        Inputs:
            length, width, height, mass: int
        Outputs:
            res: str
        """
        bulky, heavy = False, False
        L, V, M = 1e4, 1e9, 100
        if length >= L or width >= L or height >= L or length * width >= V / height:
            bulky = True
        if mass >= M:
            heavy = True
        if bulky and heavy:
            return "Both"
        elif bulky and not (heavy):
            return "Bulky"
        elif (not bulky) and heavy:
            return "Heavy"
        else:
            return "Neither"